app/src/main/java/io/github/jbellis/brokk/agents/CodeAgent.java

```diff
@@ -222,6 +230,18 @@ private TaskResult runTaskInternal(String userInput, Set<Option> options) {
                 }
             }
 
+            // PARSE-JAVA PHASE: If Java files were edited, run a parse-only check before full build
+            var parseJavaOutcome = parseJavaPhase(cs, es, metrics);
```

Comment:
This design is very language-specific, could we not introduce a linter interface of some sort?

Reply:
Lol fair enough. If that dogma works then LGTM